### PR TITLE
Put perf integration behind a feature flag

### PR DESCRIPTION
**Pull request by gaearon**
Builds on https://github.com/facebook/react/pull/11453.

The plan is to keep it `__DEV__` only for open source, but enable it based on a feature flag internally so we can experiment with using it in production for some users.

I have verified this doesn't inflate the open source build sizes.

I also exposed a ref counting interface to toggle the flag from www product code.

### CODE REVIEW

File: packages/react-cs-renderer/src/ReactNativeCSFeatureFlags.js [hunk 1]
```diff
@@ -21,6 +23,10 @@ export const enableMutatingReconciler = false;
 export const enableNoopReconciler = false;
 export const enablePersistentReconciler = true;
 
+export function addUserTimingListener() {
+  invariant(false, 'Not implemented.');
```
Comment (trueadm) [inline packages/react-cs-renderer/src/ReactNativeCSFeatureFlags.js]: Maybe a clearer message?

File: packages/shared/ReactFeatureFlags.js [hunk 2]
```diff
@@ -21,3 +23,7 @@ export const enableMutatingReconciler = true;
 export const enableNoopReconciler = false;
 // Experimental persistent mode (CS):
 export const enablePersistentReconciler = false;
+
+export function addUserTimingListener() {
+  invariant(false, 'Not implemented.');
```
Comment (trueadm) [inline packages/shared/ReactFeatureFlags.js]: Same as above

**Comment by iamdustan** [comment]
fwiw I would really like a flag to expiriment with disabling this in devmode in open source. Our performance in development on React 16 has plummeted (it’s comparable in _most_ scenarios in production with a few major regressions). The experiment is to see how much we’re paying for the usertiming integration in dev.

**Comment by gaearon** [comment]
You can disable it manually in the npm bundle and then see if it helps :-) if it does, I'm open to exposing this somehow.

**Pull request merged**